### PR TITLE
WIP: Use new xml_nodes_to_lint() in existing codebase

### PR DESCRIPTION
**Pull request by MichaelChirico**
This way we don't keep accumulating little helper functions to pull out a lint message to pass on to `xml_nodes_to_lint()`

Rename the argument `message=` ➡️  `lint_message=` to avoid writing `message <- message(expr)` which looks like `base::message()`

### CODE REVIEW

**Comment by AshesITR** [comment]
That's a rather large diff to review.
Can you first make a minimal PR and then a separate PR to refactor the `gen_*_lint` helpers?

**Comment by MichaelChirico** [comment]
SGTM. Set #960 as part 1 & rebased this relative to that.

**Comment by AshesITR** [comment]
Ah, sorry I completely overlooked the merge target.

File: R/expect_s3_class_linter.R [hunk 1]
```diff
@@ -56,13 +70,6 @@ is_s3_class_calls <- paste0("is.", c(
 ))
 
 gen_expect_s3_class_lint <- function(expr, source_file) {
```
Comment (AshesITR) [inline R/expect_s3_class_linter.R:58]: Is this still needed?

**Comment by MichaelChirico** [comment]
Not sure why I can't reopen it here... copying to new branch

**Pull request closed without merging**